Py/a_main.py

```diff
@@ -33,7 +33,7 @@ class Vars :
 
 v = Vars()
 
-bins = {
+BINS = {
```

Comment:
Suggest using the PEP8 standards for constants like this:  https://peps.python.org/pep-0008/#constants

When you used this later on, it took me a minute after combing the function a few times to realize it was not a local variable that I was just missing, but a constant defined elsewhere.